testing/btest/plugins/doctest-plugin/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+
+project(ZeekPluginDoctest)
+
+if (ENABLE_ZEEK_UNIT_TESTS)
```

Comment:
Should this and the configure script be added to the plugin skeleton?

Reply:
Yeah. Good thought to bring that up here. I had put this off since there's still an incoming round of updates to the plugin skeleton stuff, and a corresponding update over in the Zeek package template for plugins, but I don't think these are 4.1 items.

Fwiw, the test package is more crude than what I would do in an "official" version. It would be nice, for example, for a plugin to automatically build with doctest support when Zeek itself was built with it too. The demo package I mentioned in the Github issue does this, but it needed to modify configure itself, so was a bigger change.

Reply:
My only concern is to release something like this and have people use it, and then change it pretty drastically later to be a cleaner way. I guess with this coming release just being a dev release, people will understand that things may change between 4.1 and 4.2 and adapt.

Reply:
Definitely a valid concern. One thing I've been mulling over is what would be a cleaner way to determine that Zeek has doctest support built in. Over in the experimental package, I relied on `zeek --test -h` to succeed: https://github.com/ckreibich/testpackage/blob/master/configure.plugin#L30-L37

It works, but was thinking it might be nicer to teach `zeek-config` to report the build configuration. I think there's nothing that does that yet, right? Something like:

```
$ zeek-config --build_features
cpp-tests,debug,perftools
```

or some such ... and then use that.

Reply:
Generally btests do that by looking at zeek-config.h and checking what flags are enabled (see scripts/base/protocols/krb/smb2_krb.test, for example). That's kinda clunky though.

Reply:
I'm now back to wondering if we should generally compile Zeek with unit tests. As long as that's off by default, chances are that plugins won't be able to use this functionality because I imagine that almost nobody will have it available (our binary distributions don't have it, and people compiling from source need to activate it manually, which I'm sure not many do).